nucypher/blockchain/eth/agents.py

```diff
@@ -835,9 +835,17 @@ def create_policy(self,
         return receipt
 
     @contract_api(CONTRACT_CALL)
-    def fetch_policy(self, policy_id: str) -> list:
-        """Fetch raw stored blockchain data regarding the policy with the given policy ID"""
+    def fetch_policy(self, policy_id: str, with_owner=False) -> list:
+        """
+        Fetch raw stored blockchain data regarding the policy with the given policy ID.
+        If `with_owner=True`, this method executes the equivalent of `getPolicyOwner`
+        to avoid another call.
+        """
         blockchain_record = self.contract.functions.policies(policy_id).call()
+        if with_owner:
+            # If the policyOwner addr is null, we return the sponsor addr instead of the owner.
+            owner_checksum_addr = blockchain_record[1] if blockchain_record[2] == NULL_ADDRESS else blockchain_record[2]
+            return blockchain_record, owner_checksum_addr
```

Comment:
Instead of this, why not simply creating a helper function that gets the owner from a blockchain record? That way `fetch_policy()` can always only return a `list`

Reply:
We already have a `get_owner` method, but I'm trying to avoid making two separate on-chain calls.

Reply:
But this with_owner option doesn't make an additional on chain call, just reads the values in the blockchain_record list.

Reply:
Oh, I thought you were saying to get the owner from the blockchain record on-chain. I see what you're saying now.. yeah, that's def possible. I dunno how well it composes here though.